### PR TITLE
Feat/2617 program entity page connect 2 cards to data source

### PR DESCRIPTION
**Pull request by daniel-cy-lu**
# Description of changes

Removed one of the three icons and make the counts on the remaining two icons dynamic.
https://github.com/icgc-argo/platform-ui/issues/2617

## Type of Change

- [ ] Bug
- [ ] Dependency updates
- [x] Feature
- [ ] Refactoring
- [ ] Release candidate
- [ ] Styling
- [ ] Testing
- [ ] Other (please describe)

## Checklist before requesting review

- Design (select one):
  - [x] Matches design:
    - component sizes, spacing, and styles
    - font size, weight, colour
    - spelling has been double checked
  - [x] No design provided, screenshot of changes included in PR
  - [ ] No changes to UI
- Back-end (select one):
  - [ ] Back-end PRs merged, tested on develop, and linked in this PR
  - [x] No back-end changes
- [ ] Manually tested changes
- [x] Added copyrights to new files
- [x] Connected ticket to PR

Screenshots show matching results between file repo and program entity page of three Program Ids
----- ----- -----
TCRB-CA Program Entity Page Counts: **2** and **37**:

Screenshot 2023-05-12 at 3 42 35 PM

TCRB-CA File Repo Page Counts: **2** and **37**:

Screenshot 2023-05-12 at 3 42 27 PM
----- ----- -----
PACA-CA Program Entity Page Counts:

Screenshot 2023-05-12 at 3 42 15 PM

PACA-CA File Repo Counts:

Screenshot 2023-05-12 at 3 42 07 PM
----- ----- -----
RDPCDEV-CA Program Entity Page Counts:

Screenshot 2023-05-12 at 3 41 53 PM

RDPCDEV-CA File Repo Counts:

Screenshot 2023-05-12 at 3 41 44 PM

### CODE REVIEW

File: components/pages/program-entity/ProgramCardsLayout/ProgramSummaryLinkContatiner/index.tsx [hunk 1]
```diff
@@ -20,39 +20,42 @@
 import { css } from '@icgc-argo/uikit';
 
 import ProgramSummaryLink from './ProgramSummaryLink';
+import PROGRAM_SUMMARY_LINK_QUERY from '../../gql/PROGRAM_SUMMARY_LINK_QUERY';
+import { useQuery } from '@apollo/client';
+import sqonBuilder from 'sqon-builder';
+
+const ProgramSummaryLinkContainer = ({ programId }: { programId: string }) => {
+  const programFilterSqon = sqonBuilder.has('study_id', programId).build();
```
Comment (daniel-cy-lu) [inline components/pages/program-entity/ProgramCardsLayout/ProgramSummaryLinkContatiner/index.tsx:28]: build a variable with sqonBuilder. Original is too hard to read and lengthy.

File: components/pages/program-entity/ProgramCardsLayout/ProgramSummaryLinkContatiner/index.tsx [hunk 2]
```diff
@@ -20,39 +20,42 @@
...
+  const { data: { file = null } = {}, loading } = useQuery(PROGRAM_SUMMARY_LINK_QUERY, {
+    variables: {
+      SQON: programFilterSqon,
+    },
+  });
```
Comment (daniel-cy-lu) [inline components/pages/program-entity/ProgramCardsLayout/ProgramSummaryLinkContatiner/index.tsx:33]: query bucket counts of donor and file from file repo.

File: components/pages/program-entity/ProgramCardsLayout/ProgramSummaryLinkContatiner/index.tsx [hunk 3]
```diff
@@ -20,39 +20,42 @@
...
       css={css`
         width: 30%;
-        margin: 14px 0px 4px 35px;
+        margin: 35px 0px 29px 35px;
+        padding-left: 10px;
         display: flex;
         flex-direction: column;
-        justify-content: space-around;
+        justify-content: space-evenly;
       `}
```
Comment (daniel-cy-lu) [inline components/pages/program-entity/ProgramCardsLayout/ProgramSummaryLinkContatiner/index.tsx:44]: css after removing one of the three icons

File: components/pages/program-entity/ProgramCardsLayout/ProgramSummaryLinkContatiner/index.tsx [hunk 4]
```diff
@@ -20,39 +20,42 @@
...
-
-      <ProgramSummaryLink
-        circleFill={'accent2_3'}
-        iconName={'workflow'}
-        iconFill={'accent2_dark'}
-        totalNum={1039}
-        subtitle={'Donors with Clinical Data'}
+        totalNum={loading ? 'loading' : file?.aggregations.donors__donor_id.bucket_count}
+        subtitle={'Total Donors'}
       />
```
Comment (daniel-cy-lu) [inline components/pages/program-entity/ProgramCardsLayout/ProgramSummaryLinkContatiner/index.tsx:52]: removed one of the icon and make the other two dynamic by updating `totalNum`

File: components/pages/program-entity/ProgramCardsLayout/index.tsx [hunk 5]
```diff
@@ -41,9 +41,10 @@ const PaddedColumn = styled(Col)`
   align-items: stretch;
 `;
 
-const ProgramCardsLayout: ComponentType<{ programSummaryQuery: ProgramSummaryQuery }> = ({
-  programSummaryQuery,
-}) => {
+const ProgramCardsLayout: ComponentType<{
+  programSummaryQuery: ProgramSummaryQuery;
+  programId: string;
+}> = ({ programSummaryQuery, programId }) => {
```
Comment (daniel-cy-lu) [inline components/pages/program-entity/ProgramCardsLayout/index.tsx:47]: pass programId down to child

File: components/pages/program-entity/gql/PROGRAM_SUMMARY_LINK_QUERY.ts [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+        }
+        file_id {
+          bucket_count
+        }
+      }
+    }
+  }
+`;
+
+export default PROGRAM_SUMMARY_LINK_QUERY;
```
Comment (daniel-cy-lu) [inline components/pages/program-entity/gql/PROGRAM_SUMMARY_LINK_QUERY.ts:37]: gql query of file rep. result is the file and donor count of the programId.

File: components/pages/program-entity/index.tsx [hunk 7]
```diff
@@ -45,7 +45,10 @@ const ProgramEntity = ({ programId }: { programId: string }) => {
             <ProgramTitleBar programId={programId} />
           </ContentHeader>
           <ContentBody>
-            <ProgramCardsLayout programSummaryQuery={loading ? [] : program} />
+            <ProgramCardsLayout
+              programSummaryQuery={loading ? [] : program}
+              programId={programId}
+            />
```
Comment (daniel-cy-lu) [inline components/pages/program-entity/index.tsx:51]: pass programId down to child.

**Comment by demariadaniel** [comment]
Can you please add link to the ticket in the description?

**Comment by demariadaniel** [comment]
Tested + showing accurate counts for different programs w/ local UI pointing @ API Dev

**Review by demariadaniel — APPROVED** [review]

**Comment by daniel-cy-lu** [comment]
> Can you please add link to the ticket in the description?

Done. added a link in the description. If I had the ticket connected already, do I still need this in the future?

**Comment by daniel-cy-lu** [comment]
@demariadaniel Thanks Dan for checking! :D

**Comment by demariadaniel** [comment]
> > Can you please add link to the ticket in the description?
> 
> Done. added a link in the description. If I had the ticket connected already, do I still need this in the future?

Yes either way, Link in description or Connected to Issue, both work. Important because:
- Provides context for reviewer
- Provides tracking for planning
- Makes it easy to list contents included in an RC

**Pull request merged**